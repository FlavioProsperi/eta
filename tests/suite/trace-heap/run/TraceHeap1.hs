import Data.Char
import Debug.Trace

fibs :: [Integer]
fibs = 1 : 1 : zipWith (+) fibs (tail fibs)

main :: IO ()
main = do
  let transformedString = map (traceHeapId . toUpper) "Hello World!"
  traceHeapIdIO $ transformedString
  print $ take 3 transformedString
  traceHeapIdIO $ transformedString
  print $ transformedString
  traceHeapIdIO $ transformedString
  traceHeapIdIO fibs
  print $ (fibs !! 2)
  traceHeapIdIO fibs
  print $ (fibs !! 3)
  traceHeapIdIO fibs
  print $ (fibs !! 4)
  traceHeapIdIO fibs
  return ()
