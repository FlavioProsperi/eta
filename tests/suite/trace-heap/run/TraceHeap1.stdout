"HEL"
"HELLO WORLD!"
2
3
5
10946
main_transformedString[_]
x1[_] 'H'
x1[_] 'e'
x1[_] 'l'
(: 'H' (: 'E' (: 'L' (sat$4[_] 2 (unpack$2[1] [] main15[2] 4544L)))))
x1[_] 'l'
x1[_] 'o'
x1[_] ' '
x1[_] 'W'
x1[_] 'o'
x1[_] 'r'
x1[_] 'l'
x1[_] 'd'
x1[_] '!'
['H', 'E', 'L', 'L', 'O', ' ', 'W', 'O', 'R', 'L', 'D', '!']
(: 1 (: 1 main5[_]))
(: 1 (:#1 1 #1(: 2 (sat[_] @:#1 @main5#1))))
(: 1 (: 1 #1(: 2 #1(: 3 (sat[_] @main5#1 @sat#1)))))
(: 1 (: 1 (: 2 #1(: 3 #2(: 5 (sat[_] @sat#1 @sat#2))))))
(: 1 (: 1 (: 2 (: 3 (: 5 (: 8 (: 13 (: 21 (: 34 (: 55 (: 89 (: 144 (: 233 (: 377 (: 610 (: 987 (: 1597 (: 2584 (: 4181 #1(: 6765 #2(: 10946 (sat[_] @sat#1 @sat#2))))))))))))))))))))))
